### ISSUE TITLE
Portfolio user can view/edit domains based on their permissions

### ISSUE THREAD

**Issue by vickyszuchin** [cisagov/manage.get.gov#2350]
### Issue description

The portfolio user can view domains in the portfolio. The ability to **manage** the domains in a portfolio is restricted. An user can manage a domain (edit domain information) if they are defined as a domain manager of the domain. Otherwise, the user can view the domain information but can not edit the domain information

### Acceptance criteria

- [ ] only domains in the user's portfolio are shown on the Domains page. If the user has a domain that is NOT part of their portfolio, it will NOT show. (this user would need to reach out to us about a missing domain)
- [ ] A portfolio user **can** edit a domain in the portfolio only if they are defined as a domain manager (note the manage icon should appear in this case)
- [ ] admins and read-only admins can see all domains in their org in this table.
- [ ] admins and read-only admins cannot edit a domain UNLESS they are added as a domain manager to a particular domain. They should see the eye icon (view only) on all domains that they don't have a UserDomainRole for. (Note, the portfolio read-only view that shows when the eye icon is clicked will be made in #2379)
- [ ] The same state rules apply that can make a domain view only (eye icon appears on the overview page), such as Deleted and on hold domains still need to show as view only
- [ ] The domains page/ tab is only shown to those with domain management permissions, read-only admin and administrator privileges  all others would get a 404 page not found when going to this page

### Additional context

Note that this heavily relates to #2379 both could be done in the same PR if desired. 

### Links to other issues
- blocked by #2366 
- related to #2367 
- blocks #2379
